### ISSUE TITLE
Plugins

### ISSUE THREAD

**Issue by yatharthk** [developit/snarkdown#19]
I was wondering if we can make `snarkdown` pluggable. So the core module remains minimal and developers can use plugins to enhance the conversion as and how they need. What do you say @developit ?

**Comment by tunnckoCore** [comment]
It was commeted here a bit https://github.com/developit/snarkdown/issues/5

**Comment by developit** [comment]
Yup. The difficulty here is that the lexer is a regex. It's very hard to dynamically build up a regex and still have meaningful submatches.
